core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -776,6 +781,42 @@ class UserService(
       } yield {}
     }
 
+  def batchUpdateBillingProjectMembers(projectName: RawlsBillingProjectName,
+                                       batchProjectAccessUpdate: BatchProjectAccessUpdate,
+                                       inviteUsersNotFound: Boolean
+  ): Future[Unit] =
+    requireProjectAction(projectName, SamBillingProjectActions.alterPolicies) {
+      val membersToAdd = batchProjectAccessUpdate.membersToAdd
+      val membersToRemove = batchProjectAccessUpdate.membersToRemove
+
+      collectMissingUsers(membersToAdd.map(_.email), ctx) flatMap { membersToInvite =>
+        if (membersToInvite.isEmpty || inviteUsersNotFound) {
+          for {
+            invites <- Future.traverse(membersToInvite) { invite =>
+              samDAO.inviteUser(invite, ctx).map { _ =>
+                Notifications.BillingProjectInvitedNotification(
+                  WorkbenchEmail(invite),
+                  WorkbenchUserId(ctx.userInfo.userSubjectId.value),
+                  projectName.value
+                )
+              }
+            }
+            additions <- Future.traverse(membersToAdd) { projectAccessUpdate =>
+              addUserToBillingProjectV2(projectName, projectAccessUpdate)
+            }
+            removals <- Future.traverse(membersToRemove) { projectAccessUpdate =>
+              removeUserFromBillingProjectV2(projectName, projectAccessUpdate)
+            }
+          } yield notificationDAO.fireAndForgetNotifications(invites)
+        } else
+          Future.failed(
+            new RawlsExceptionWithErrorReport(
+              ErrorReport(StatusCodes.Conflict, s"Users ${membersToInvite.mkString(",")} have not signed up for Terra")
```

Comment:
Not for this PR, but: I would like to get away from rendering presentation layer language in errors for consumption by the UI and instead render the error into a semantically meaningful structure that the UI (or really any api consumer like a cli tool) can then consume and render however it chooses. Something like this:
```
409
{
"missingUsers": ["foo@bar.com", "bar@baz.com"]
}
```